src/lib/util.ts

```diff
@@ -297,3 +328,35 @@ export function withoutArticle(name: string): string {
 export function excludeArticleSort(a: string, b: string): number {
 	return withoutArticle(a).localeCompare(withoutArticle(b));
 }
+
+export function getSteamID(str: string): string {
+	let trimmed = str.trim();
+	if (isOnlyDigits(trimmed)) return trimmed;
+
+	let url: URL | null = null;
+	try {
+		url = new URL(trimmed);
+	} catch (e: any) {
+		return '';
+	}
+
+	if (url?.hostname !== 'steamcommunity.com') return '';
+
+	let id = url?.searchParams?.get('id');
+	if (id === null) return '';
+
+	if (isOnlyDigits(id)) return id;
+
+	id = id.substring(0, id.search(/[^0-9]/));
+	if (isOnlyDigits(id)) return id;
+
+	return '';
+}
+
+export function validateSteamID(str: string): string | boolean {
```

Comment:
This doesn't return a boolean?
```suggestion
export function validateSteamID(str: string): string {
```